File: MainApp/src/main/java/com/sleticalboy/dailywork/accounts/auth/AuthenticationService.java
package com.sleticalboy.dailywork.accounts.auth;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;

import androidx.annotation.Nullable;

import com.sleticalboy.dailywork.accounts.auth.Authenticator;

/**
 * Created by AndroidStudio on 20-2-23.
 *
 * @author binlee
 */
public final class AuthenticationService extends Service {

    private static final String TAG = "AuthenticationService";

    private Authenticator mAuthenticator;

    @Override
    public void onCreate() {
        mAuthenticator = new Authenticator(this);
        Log.d(TAG, "onCreate() called");
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        Log.d(TAG, "onBind() called with: intent = [" + intent + "]");
        return mAuthenticator.getIBinder();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy() called");
    }
}

File: app/src/main/java/com/binlee/learning/accounts/auth/AuthenticationService.kt
package com.binlee.learning.accounts.auth

import android.app.Service
import android.content.Intent
import android.os.IBinder

/**
 * Created by AndroidStudio on 20-2-23.
 *
 * @author binlee
 */
class AuthenticationService : Service() {

  private var mAuthenticator: Authenticator? = null

  override fun onCreate() {
    mAuthenticator = Authenticator(this)
    // Log.d(TAG, "onCreate() called");
  }

  override fun onBind(intent: Intent): IBinder? {
    // Log.d(TAG, "onBind() called with: intent = [" + intent + "]");
    return mAuthenticator!!.iBinder
  }

  companion object {
    private const val TAG = "AuthenticationService"
  }
}